airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -2773,6 +2773,14 @@ def _submit_job(
         hook: BigQueryHook,
         job_id: str,
     ) -> BigQueryJob:
+        # Annotate the job with dag and task id labels
+        if "labels" in self.configuration:
+            if isinstance(self.configuration["labels"], dict):
+                self.configuration["labels"]["airflow-dag"] = self.dag_id
+                self.configuration["labels"]["airflow-task"] = self.task_id
```

Comment:
We should do a pattern match here so that no character that is not allowed in `configuration` goes in

Reply:
Added a pattern match